File: Python/iniciante/1012area.py
linha = input().split(" ")
a, b, c = linha
a = float(a)
b = float(b)
c = float(c)
pi = 3.14159
triangulo = (a * c) / 2
circulo = (c ** 2) * pi
trapezio = ((a + b) / 2) * c
quadrado = b ** 2
retangulo = a * b
print("TRIANGULO: %.3f\n"
      "CIRCULO: %.3f\n"
      "TRAPEZIO: %.3f\n"
      "QUADRADO: %.3f\n"
      "RETANGULO: %.3f" % (triangulo, circulo, trapezio, quadrado, retangulo))

File: Haskell/Iniciante/1012area.hs
import           Text.Printf

wordsWhen :: (Char -> Bool) -> String -> [String]
wordsWhen p s = case dropWhile p s of
      "" -> []
      s' -> w : wordsWhen p s'' where (w, s'') = break p s'

main :: IO ()
main = do
      linha <- getLine
      let list = wordsWhen (== ' ') linha
      let a    = read (head list) :: Double
      let b    = read (list !! 1) :: Double
      let c    = read (list !! 2) :: Double
      printf "TRIANGULO: %.3f\n" ((a * c) / 2)
      printf "CIRCULO: %.3f\n"   ((c ** 2) * 3.14159)
      printf "TRAPEZIO: %.3f\n"  (((a + b) / 2) * c)
      printf "QUADRADO: %.3f\n"  (b ** 2)
      printf "RETANGULO: %.3f\n" (a * b)
      return ()
